onyo/lib/tests/test_commands_cat.py

```diff
@@ -0,0 +1,118 @@
+import pytest
+
+from onyo.lib.inventory import Inventory
+from onyo.lib.onyo import OnyoRepo
+from ..assets import Asset
+from ..commands import onyo_cat
+from ... import OnyoInvalidRepoError
+
+
+@pytest.mark.ui({'yes': True})
+def test_onyo_cat_errors(inventory: Inventory) -> None:
+    """`onyo_cat` must raise the correct error in different illegal or impossible calls."""
+    asset_path = inventory.root / "somewhere" / "nested" / "TYPE_MAKER_MODEL.SERIAL"
+    dir_path = inventory.root / 'empty'
+
+    # cat with no file
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[])
+
+    # cat on dir
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[dir_path])
+
+    # cat on non-existing file
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[dir_path / "does_not_exi.st"])
+
+    # cat on untracked file
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[inventory.root / "untracked" / "file"])
+
+    # outside of onyo repository
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[inventory.root / ".."])
+
+    # one of multiple is non-existing
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[asset_path,
+                         inventory.root / "doesnotexist",
+                         asset_path])
+
+    # cat on .anchor
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[dir_path / OnyoRepo.ANCHOR_FILE_NAME])
+
+    # cat on a template file
+    pytest.raises(ValueError,
+                  onyo_cat,
+                  inventory,
+                  paths=[inventory.repo.dot_onyo / "templates" / "laptop.example"])
+
+    # cat on a file that is in the inventory, but has invalid YAML contents
+    invalid_path = inventory.root / "in_va_l.id"
+    invalid_path.touch()
+    invalid_path.write_text("key: value\ninvalid")
+    inventory.repo.git._git(["add", str(invalid_path)])
+    inventory.commit("Invalid file added")
+    pytest.raises(OnyoInvalidRepoError,
+                  onyo_cat,
+                  inventory,
+                  paths=[invalid_path])
+
+
+@pytest.mark.ui({'yes': True})
+def test_onyo_cat_single(inventory: Inventory) -> None:
+    """`onyo_cat()` a single valid asset."""
+    asset_path = inventory.root / "somewhere" / "nested" / "TYPE_MAKER_MODEL.SERIAL"
+
+    # cat single asset
+    onyo_cat(inventory,
+             paths=[asset_path])
+
```

Comment:
Here you only test that cat does not error, but not what it does.
One could use pytest's `capsys` fixture to check the file's content is put on stdout.

Same for the variants below.